v3/src/components/axis/components/sub-axis.tsx

```diff
@@ -5,6 +5,7 @@ import {useAxisProviderContext} from "../hooks/use-axis-provider-context"
 import {useSubAxis} from "../hooks/use-sub-axis"
 import {isNumericAxisModel} from "../models/axis-model"
 import {NumericAxisDragRects} from "./numeric-axis-drag-rects"
+import { useGraphContentModelContext } from "../../graph/hooks/use-graph-content-model-context"
```

Comment:
@kswenson The latest update improves things in terms of keeping graph-specific references out of the axis world although it's not clear to me that it's as solid as it could be. Most specifically, the new `nonDraggableAxisTicks` method I added in addition to `hasDraggableNumericAxis` seems a little questionable. 

I believe the tick format used for the binned plot is unique to the graph and also requires access to information about the bin width and total number of bins which is currently computed in the graph's data configuration. So either the graph has to generate the tick values and labels, or else the axis needs to access those bin details. It's not clear to me which is better, but I opted for having the graph do most of the work. It's possible I'm missing other options, but I suspect anything else may require a more significant change to the structure of things?

Note too that these changes seem to require that I add `hasDraggableNumericAxis` and `nonDraggableAxisTicks` to `kDefaultDataDisplayModel` in use-data-display-model.ts. Not doing that leads to the errors `hasDraggableNumericAxis is not a function` (which we were seeing last week) and `nonDraggableAxisTicks is not a function`.

Reply:
Your approach seems reasonable and I don't think it's worth spending more time on it at this point.